package.json

```diff
@@ -119,8 +123,10 @@
     "@types/lodash.get": "^4.4.6",
     "@types/lodash.isequal": "^4.5.2",
     "@types/lodash.set": "^4.3.6",
+    "@types/mocha": "^9.0.0",
```

Comment:
There's some type collisions here inside node_modules (across globals declared by jest and mocha) that I need help dealing with.

Reply:
is the problem that we have jest and mocha tests in the same namespace? We could use project references to split cli unit tests up in that case. Or is using jest-mock/snapshots/expect enough to clash with the mocha types?

Reply:
They both do toplevel `declare var`, so the global namespace.

I don't see how project references would resolve it–rather, I don't know enough about using projects in TS to know how it'd solve things.

Reply:
`const {describe, it} = require('mocha')` works (though the globals still exist), so we could also convert to using actual imports rather than the terrible global thing all these test runners do. This person apparently keeps a fork of `@types/mocha` but with only local declarations: https://github.com/mochajs/mocha/issues/956#issuecomment-665043713 🤷 

But it might just be easier to use project references.

> I don't see how project references would resolve it–rather, I don't know enough about using projects in TS to know how it'd solve things.

Really they wouldn't even be project references here, it would just be a different tsconfig which doesn't include the jest types (or the mocha types, depending on where we put it). This is the same as e.g. the treemap tsconfig, where we only include the `tabulator-tables` types from `node_modules`, and exclude all the other types (like we don't want any of the `@types/node` globals muddying up code suggestions in code that's only going to run in the browser):

https://github.com/GoogleChrome/lighthouse/blob/82772f93645989c9c250c78a0ad10d55f3372beb/treemap/tsconfig.json#L8-L9

FWIW projects references are just that, multiple tsconfigs, with the addition of the `references` section where one tsconfig can reference another as a _type_ dependency, e.g. treemap pulling in `'../report/'` as a dependency in the exact same way it's pulling in `tabulator-tables` as one. It's much more analogous to our multiple eslintrc files than yarn/npm workspaces or lerna or whatever.

Reply:
I've run into complications with this approach because referencing files outside of a tsconfig project is not allowed.

![image](https://user-images.githubusercontent.com/4071474/152875647-4bb9ed06-0b55-4758-93cc-c9524d741e79.png)

```diff
diff --git a/lighthouse-cli/test/cli/mocha-globals.d.ts b/lighthouse-cli/test/cli/mocha-globals.d.ts
new file mode 100644
index 000000000..23f979f1f
--- /dev/null
+++ b/lighthouse-cli/test/cli/mocha-globals.d.ts
@@ -0,0 +1,5 @@
+import expect_ from 'expect';
+
+declare global {
+  const expect: typeof expect_;
+}
diff --git a/lighthouse-cli/test/cli/tsconfig.json b/lighthouse-cli/test/cli/tsconfig.json
new file mode 100644
index 000000000..88fce8081
--- /dev/null
+++ b/lighthouse-cli/test/cli/tsconfig.json
@@ -0,0 +1,15 @@
+{
+  "extends": "../../../tsconfig-base.json",
+  "compilerOptions": {
+    // "rootDir": "../../..",
+    // "outDir": "../../../.tmp/tsbuildinfo/cli-tests",
+    // Selectively include types from node_modules/.
+    "types": ["tabulator-tables", "mocha"],
+  },
+  "include": [
+    "**/*.js",
+    "mocha-globals.d.ts",
+    "../../../types/global-lh.d.ts",
+    // "../../../**/*.js"
+  ],
+}

```